cli/lib/rest/api.c

```diff
@@ -319,30 +319,30 @@ kvdb_get_kvs_names_cb(
     assert(cJSON_IsArray(body));
 
     len = cJSON_GetArraySize(body);
-    buf = calloc(len, sizeof(*namev) + HSE_KVS_NAME_LEN_MAX);
-    if (!buf) {
+    namev = calloc(len, sizeof(*namev) + HSE_KVS_NAME_LEN_MAX);
+    if (!namev) {
         err = merr(ENOMEM);
         goto out;
     }
 
-    namev = &buf;
+    name = (char *)&namev[len];
```

Comment:
confusing..   isn't  'name = namev[len];'  the same thing?

Reply:
Maybe. Let me test. Pulled this out of the kvs_names logic in ikvdb.c

Reply:
nevermind..

Reply:
ah, i would prob write this as  'name = (char *)(name + len);', but either or...

Reply:
apparently it isn't the same thing

Reply:
baffling...  not sure what's going on there...

Reply:
It is

[ptr ptr..., HSE_KVS_NAME_LEN_MAX HSE_KVS_NAME_LEN_MAX...]